docs/first_steps.md

```diff
@@ -123,59 +120,96 @@ The first parameter given to the constructor is the first element of the domain,
+
+Finally, we determine the number of time steps and as we did with the `X` and `Y` dimensions, we create the time domain. 
+
+\snippet uniform_heat_equation.cpp time-domain
 
 
 # Data allocation
 
 We allocate two 2D Chunks along the X and Y dimensions which will be used to map temperature to the domains' points at t and t+dt. When constructing the Chunks one can give an optional string to label the memory allocations. This helps debugging and profiling applications using the Kokkos tools, see also [Kokkos Tools](https://github.com/kokkos/kokkos-tools).
 
-\snippet heat_equation.cpp data allocation
+These chunks map the temperature into the full domain (including ghosts) twice:
++ *ghosted_last_temp* for the last fully computed time-step.
++ *ghosted_next_temp* for time-step being computed.
+
+\snippet uniform_heat_equation.cpp data allocation
 
+Note that the `DeviceAllocator` is responsible for allocating memory on the default memory space.
 
 # Initial conditions
 
-\snippet heat_equation.cpp initial-conditions
+To set the initial conditions, the `ghosted_intial_temp` is created and acts as a pointer to the chunk. The const qualifier makes it clear that ghosted_initial_temp always references the same chunk, `ghosted_last_temp` in this case.
 
-\snippet heat_equation.cpp initial output
+\snippet uniform_heat_equation.cpp initial-chunkspan
 
+Then, we iterate over each *DiscreteElement* of the domain to fill `ghosted_initial_temp` with the initial values of the simulation.
+
+\snippet uniform_heat_equation.cpp fill-initial-chunkspan
+
+To display the data, a chunk is created on the host.
+
+\snippet uniform_heat_equation.cpp host-chunk
+
+We deepcopy the data from the `ghosted_last_temp` chunk to `ghosted_temp` on the host.
+
+\snippet uniform_heat_equation.cpp initial-deepcopy
+
+And we display the initial data.
+
+\snippet uniform_heat_equation.cpp initial-display
 
 # Time loop
 
-\snippet heat_equation.cpp time iteration
+\snippet uniform_heat_equation.cpp time iteration
 
 
 ## Periodic conditions
 
-\snippet heat_equation.cpp boundary conditions
+\snippet uniform_heat_equation.cpp boundary conditions
 
 
 ## Numerical scheme
 
-\snippet heat_equation.cpp manipulated views
+For the numerical scheme, two chunkspans are created: 
++ `next_temp` a span excluding ghosts of the temperature at the time-step we will build.
++ `last_temp` a read-only view of the temperature at the previous time-step.Note that *span_cview* returns a read-only ChunkSpan.
+
+\snippet uniform_heat_equation.cpp manipulated views
+
+We then solve the equation.
 
-\snippet heat_equation.cpp numerical scheme
+\snippet uniform_heat_equation.cpp numerical scheme
 
 
 ## Output
 
-\snippet heat_equation.cpp output
+\snippet uniform_heat_equation.cpp output
 
 
 ## Final swap
 
-\snippet heat_equation.cpp swap
+\snippet uniform_heat_equation.cpp swap
 
 
 # Final output
 
-\snippet heat_equation.cpp final output
+\snippet uniform_heat_equation.cpp final output
```

Comment:
If we have nothing to say, shall we keep it ?

Reply:
For me, we should not keep it. I think it adds nothing to the understanding of DDC